### DISCUSSION TITLE
Running SolidJS with NativeScript

### DISCUSSION THREAD

**Discussion by ammarahm-ed** [solidjs/solid#1214] [Q&A]
Hi, I have been trying to run solid with NativeScript using [dominative](https://github.com/SudoMaker/DOMiNATIVE) as a dom. Although everything compiles correctly I keep getting xx function not found errors as soon as the app launches on emulator. It would be great if you can guide me to make it work with webpack so that it compiles solid properly without any missing functions.

Here's the repo: https://github.com/ammarahm-ed/solidjs-native

<details>

<summary>webpack.config.js</summary>

```js
const webpack = require("@nativescript/webpack");
const {
  getPlatformName,
} = require("@nativescript/webpack/dist/helpers/platform");

module.exports = (env) => {
  webpack.init(env);
  webpack.useConfig("base");
  const config = solid(webpack.resolveChainableConfig(), env);
  return config.toConfig();
};

/**
 *
 * @param {import('webpack-chain')} config
 * @param {*} env
 * @returns
 */
function solid(config, env) {
  const platform = getPlatformName();
  const mode = env.production ? "production" : "development";
  const production = mode === "production";

  config.resolve.extensions
    .prepend(".tsx")
    .prepend(`.${platform}.tsx`)
    .prepend(`.${platform}.ts`);
  config.output.devtoolNamespace("app");

  // I have tried to disable these but still getting same issue
  config.optimization.minimize(false);
  config.optimization.usedExports(false);
  config.optimization.providedExports(false);

  config.optimization.end();

  config.module
    .rule("bundle-source")
    .test(/\.(|t|j)sx?$/)
    .exclude.add(/node_modules/)
    .end()
    .use("babel-loader")
    .loader("babel-loader")
    .before("ts-loader")
    .options({
      babelrc: false,
      configFile: false,
      presets: ["solid"],
    });

  return config;
}

```

</details>

<details>

<summary>app.tsx</summary>

```js
import { Application } from "@nativescript/core";
import "./dom.js";
import { createSignal, onCleanup, Component } from "solid-js";
import { render, template } from "solid-js/web/dist/web";
console.log("render is a", render, template);

const App = () => {
  const [count, setCount] = createSignal(0),
    timer = setInterval(() => setCount((c) => c + 1), 1000);
  onCleanup(() => clearInterval(timer));
  return <div>{count()}</div>;
};

Application.run({
  //@ts-ignore
  create: () => {
    const root = document.createElement("FlexboxLayout");
    document.appendChild(root);
    //@ts-ignore
    root.tagName = "FlexboxLayout";
    //@ts-ignore
    document.activeElement = root;
    //@ts-ignore
    render(App, root);
    return root;
  },
});

```
</details>

<details>

<summary>tsconfig.json</summary>

```js
{
  "compilerOptions": {
    "module": "commonjs",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "noEmitHelpers": true,
    "noEmitOnError": false,
    "skipLibCheck": true,
    "noEmit": true,
    "maxNodeModuleJsDepth": 5,
    "downlevelIteration": true,
    "jsx": "preserve",
    "jsxImportSource": "solid-js",
    "esModuleInterop": true,
    "sourceMap": true,
    "allowJs": true,
    "lib": ["es6", "dom"],
    "baseUrl": "."
  },
  "files": [
    "./references.d.ts",
    "./src/solid-native-jsx.ts",
    "./src/nativescript-jsx.ts"
  ]
}

```

</details>

<details>

<summary>Error</summary>

System.err: An uncaught Exception occurred on "main" thread.
  System.err: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Error 
  calling module function
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err: File: (file: src/webpack:/app/webpack/bootstrap:27:0)
  System.err:
  System.err: StackTrace:
  System.err: ./src/app.tsx(file: src/webpack:/app/src/app.tsx:1:28)
  System.err:   at __webpack_require__(file: src/webpack:/app/webpack/bootstrap:24:0)
  System.err:   at __webpack_exec__(file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:207:39)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:221)
  System.err:   at __webpack_require__.X(file: src/webpack:/app/webpack/runtime/startup entrypoint:6:0)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:47)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:213:3)
  System.err:   at require(:1:266)
  System.err:
  System.err:
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err:
  System.err: StackTrace:
  System.err: java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.N
  ativeScriptException: Error calling module function
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err: File: (file: src/webpack:/app/webpack/bootstrap:27:0)
  System.err:
  System.err: StackTrace:
  System.err: ./src/app.tsx(file: src/webpack:/app/src/app.tsx:1:28)
  System.err:   at __webpack_require__(file: src/webpack:/app/webpack/bootstrap:24:0)
  System.err:   at __webpack_exec__(file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:207:39)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:221)
  System.err:   at __webpack_require__.X(file: src/webpack:/app/webpack/runtime/startup entrypoint:6:0)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:47)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:213:3)
  System.err:   at require(:1:266)
  System.err:
  System.err:
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err:   at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6730)
  System.err:   at android.app.ActivityThread.access$1500(ActivityThread.java:247)
  System.err:   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2053)
  System.err:   at android.os.Handler.dispatchMessage(Handler.java:106)
  System.err:   at android.os.Looper.loopOnce(Looper.java:201)
  System.err:   at android.os.Looper.loop(Looper.java:288)
  System.err:   at android.app.ActivityThread.main(ActivityThread.java:7839)
  System.err:   at java.lang.reflect.Method.invoke(Native Method)
  System.err:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  System.err:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
  System.err: Caused by: com.tns.NativeScriptException: Error calling module function
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err: File: (file: src/webpack:/app/webpack/bootstrap:27:0)
  System.err:
  System.err: StackTrace:
  System.err: ./src/app.tsx(file: src/webpack:/app/src/app.tsx:1:28)
  System.err:   at __webpack_require__(file: src/webpack:/app/webpack/bootstrap:24:0)
  System.err:   at __webpack_exec__(file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:207:39)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:221)
  System.err:   at __webpack_require__.X(file: src/webpack:/app/webpack/runtime/startup entrypoint:6:0)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:208:47)
  System.err:   at (file:///data/data/org.nativescript.ReactDomExp/files/app/bundle.js:213:3)
  System.err:   at require(:1:266)
  System.err:
  System.err:
  System.err: TypeError: (0 , solid_js_web__WEBPACK_IMPORTED_MODULE_1__.template) is not a function
  System.err:   at com.tns.Runtime.runModule(Native Method)
  System.err:   at com.tns.Runtime.runModule(Runtime.java:689)
  System.err:   at com.tns.Runtime.run(Runtime.java:681)
  System.err:   at com.tns.NativeScriptApplication.onCreate(NativeScriptApplication.java:21)
  System.err:   at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1211)
  System.err:   at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6725)
  System.err:   ... 9 more

</details

Previously I have been able to run `react` with `dominative` & `nativescript` so I think it should be possible to do it with SolidJS too without using a custom renderer.

**Comment by ryansolid** [comment]
It is probably getting the server version of the runtime. Is there a way to set export conditions? It must be resolving to node. https://webpack.js.org/guides/package-exports/#conditions. These are a very powerful mechanism and they are built into a lot tooling which is powerful but it is also has potential of stepping on landmines. You could try "browser" condition but that may have other side effects.

**Reply by ammarahm-ed** [reply]
@ryansolid thanks for the quick response. I will try that out and see what happens
